From de79eaea0f12276543f74bf03183fe1d63c559e4 Mon Sep 17 00:00:00 2001
From: Matthijs van Duin <matthijsvanduin@gmail.com>
Date: Mon, 5 Oct 2015 04:29:56 +0200
Subject: [PATCH 25/25] nvmem: fix permissions of readonly nvmem binattr

Hi Robert, Srinivas,

With the current -bone kernels and my own dtb I found these weird perms:

---------x 1 root root 4097 Oct  5 03:25 /sys/bus/nvmem/devices/at24-0/nvmem

I traced the problem to a bug in nvmem triggered by the fact I had
marked the EEPROM readonly (since it is).

Patch attached.

-- >8 --

Signed-off-by: Matthijs van Duin <matthijsvanduin@gmail.com>
---
 drivers/nvmem/core.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/nvmem/core.c b/drivers/nvmem/core.c
index ef4aa7f..95e0132 100644
--- a/drivers/nvmem/core.c
+++ b/drivers/nvmem/core.c
@@ -311,7 +311,7 @@ struct nvmem_device *nvmem_register(const struct nvmem_config *config)
 	nvmem->read_only = of_property_read_bool(np, "read-only") |
 			   config->read_only;
 	nvmem->bin = bin_attr_template;
-	nvmem->bin.attr.mode = nvmem->read_only ? : (S_IRUSR | S_IWUSR);
+	nvmem->bin.attr.mode = nvmem->read_only ? S_IRUSR : (S_IRUSR | S_IWUSR);
 	nvmem->bin.attr.mode |= config->perm;
 	nvmem->bin.size = nvmem->size;
 	device_initialize(&nvmem->dev);
-- 
2.6.2

